pkg/framework/machinesets.go

```diff
@@ -285,6 +317,40 @@ func GetWorkerMachineSets(ctx context.Context, client runtimeclient.Client) ([]*
 	return result, nil
 }
 
+// getArchitectureFromMachineSetNodes returns the architecture of the nodes controlled by the given machineSet's machines.
+func getArchitectureFromMachineSetNodes(ctx context.Context, client runtimeclient.Client, machineSet *machinev1.MachineSet) (string, error) {
+	machines, err := GetMachines(ctx, client)
+	if err != nil {
+		return "", fmt.Errorf("error getting machines: %w", err)
+	}
+
+	for key := range machines {
+		if metav1.IsControlledBy(machines[key], machineSet) {
+			// Get the node
+			if node, err := getNode(ctx, client, machines[key].Status.NodeRef.Name); err != nil {
+				// If we can't get a node controlled by the first machine owned by the machineSet, return an error
+				return "", fmt.Errorf("error getting node: %w", err)
+			} else {
+				return node.Status.NodeInfo.Architecture, nil
+			}
+		}
+	}
+
+	return "", errMachineSetHasNoNodes
+}
+
+// getNode returns the corev1.Node object given the name.
+func getNode(ctx context.Context, client runtimeclient.Client, name string) (corev1.Node, error) {
```

Comment:
i don't think we need to use this function, we have utility functions to get a Node from a Machine or MachineSet, see https://github.com/openshift/cluster-api-actuator-pkg/blob/master/pkg/framework/nodes.go#L80-L121